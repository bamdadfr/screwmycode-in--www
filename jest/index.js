export { JestWithStyledComponents } from './jest-with-styled-components'