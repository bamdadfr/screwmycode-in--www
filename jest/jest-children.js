import React from 'react'

/**
 * @function
 * @name JestChildren
 * @description jest: dummy children
 * @returns {React.ReactNode} - react component
 */
export function JestChildren () {

    return (
        <>
            <span>children</span>
            <span>children</span>
            <span>children</span>
        </>
    )

}