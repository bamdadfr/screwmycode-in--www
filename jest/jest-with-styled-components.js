import React from 'react'
import { ThemeProvider } from 'styled-components'
import { Global, Theme } from '../src/app/styles'

/**
 * @function
 * @name JestWithStyledComponents
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactElement} Component - react sub component
 * @returns {React.ReactElement} - react component
 */
export function JestWithStyledComponents (Component) {

    return (
        <>
            <Global/>
            <ThemeProvider theme={Theme}>
                {Component}
            </ThemeProvider>
        </>
    )

}