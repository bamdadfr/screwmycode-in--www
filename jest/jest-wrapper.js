import React from 'react'
import { RecoilRoot } from 'recoil'
import { ThemeProvider } from 'styled-components'
import { ThemeStyles } from '../src/styles/theme.styles'

/**
 * @function
 * @name JestWrapper
 * @description jest: wrapper with recoil and styled-components for tests
 * @param {React.ReactNode} Component - react sub component
 * @returns {React.ReactNode} - react component
 */
export function JestWrapper (Component) {

    return (
        <RecoilRoot>
            <ThemeProvider theme={ThemeStyles}>
                {Component}
            </ThemeProvider>
        </RecoilRoot>
    )

}