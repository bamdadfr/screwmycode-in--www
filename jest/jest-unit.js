import React from 'react'
import { shallow } from 'enzyme'
import { JestWithStyledComponents } from './jest-with-styled-components'

/**
 * @function
 * @name JestUnit
 * @description jest: get unit from enzyme
 * @param {React.ReactElement} Component - react component
 * @param {boolean} [wrap=false] - wrap with state and theme...
 * @returns {object} - enzyme shallow object
 */
export function JestUnit (Component, wrap = false) {

    if (wrap) {

        return shallow (
            JestWithStyledComponents (Component),
        )

    }

    return shallow (Component)

}