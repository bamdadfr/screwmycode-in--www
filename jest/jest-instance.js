import renderer from 'react-test-renderer'
import { JestWrapper } from './jest-wrapper'

export const JestInstance = (Component) => renderer.create (JestWrapper (Component)).toJSON ()