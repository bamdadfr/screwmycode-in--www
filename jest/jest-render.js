import React from 'react'
import { render } from '@testing-library/react'
import { JestWrapper } from './jest-wrapper'

/**
 * @function
 * @name JestRender
 * @description jest: get a render from testing-library
 * @param {React.ReactElement} Component - react component to render
 * @returns {object} - testing library render
 */
export function JestRender (Component) {

    return render (
        JestWrapper (Component),
    )

}