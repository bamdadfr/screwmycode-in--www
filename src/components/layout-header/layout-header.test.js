import React from 'react'
import renderer from 'react-test-renderer'
import LayoutHeaderComponent from '@/components/layout-header/layout-header.component'
import { JestWrapper } from '../../../jest/jest-wrapper'

describe ('LayoutHeaderComponent', () => {

    const Component = <LayoutHeaderComponent/>
    const instance = renderer.create (JestWrapper (Component)).toJSON ()

    it ('should match snapshot', () => {

        expect (instance).toMatchSnapshot ()

    })

    it ('should have 2 children', () => {

        expect (instance.children).toHaveLength (2)

    })

})