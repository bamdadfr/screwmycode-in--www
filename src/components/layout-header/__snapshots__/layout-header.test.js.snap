// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutHeaderComponent should match snapshot 1`] = `
<div
  className="sc-1yrrc8c-0 fUerQO"
>
  <a
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M10 20v-6h4v6h5v-8h3L12 3L2 12h3v8h5z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </a>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={-1}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M17 17H7v-3l-4 4l4 4v-3h12v-6h-2M7 7h10v3l4-4l-4-4v3H5v6h2V7z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </span>
</div>
`;
